### ISSUE TITLE
[promise-function-async] Error line offset when using decorators

### ISSUE THREAD

**Issue by RecuencoJones** [typescript-eslint/typescript-eslint#751]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "tsconfig.json"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "@typescript-eslint/promise-function-async": "error"
  }
}
```

```JSON
{
  "compilerOptions": {
    "target": "es2017",
    "experimentalDecorators": true
  },
  "files": [
    "index.ts"
  ]
}
```

```TS
const SomeFancyDecorator = (): MethodDecorator => (target) => target;

export class Foo {
  @SomeFancyDecorator()
  method() {
    return Promise.resolve('foo');
  }
}
```

**Expected Result**

```
/Users/recuedav/Workspace/GitHub/RecuencoJones/typescript-eslint-issue/index.ts
  5:3  error  Functions that return promises must be async  @typescript-eslint/promise-function-async

✖ 1 problem (1 error, 0 warnings)
```

**Actual Result**

```
/Users/recuedav/Workspace/GitHub/RecuencoJones/typescript-eslint-issue/index.ts
  4:3  error  Functions that return promises must be async  @typescript-eslint/promise-function-async

✖ 1 problem (1 error, 0 warnings)
```

**Additional Info**

Repository for reproduction: https://github.com/RecuencoJones/typescript-eslint-issue-751

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.11.0` |
| `@typescript-eslint/parser`        | `1.11.0` |
| `TypeScript`                       | `3.4.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.7.0` |

**Comment by bradzacher** [comment]
This is working as expected.

As far as the AST is concerned (and typescript), the decorator is considered to be _owned_ by the method, and as such it is included within its bounds.

<details>
<summary>AST Example</summary>

Note in the following example that the method definition has the `decorator` property.

[repl](https://astexplorer.net/#/gist/b6de73c6c4a9092d06325010d94ffa8d/8a63fe35030a6277195f40f452a85bee85551082)
```
{
    "type": "MethodDefinition",
    "key": {
        "type": "Identifier",
        "name": "method",
        "range": [116, 122],
        "loc": {
            "start": {
                "line": 5,
                "column": 2
            },
            "end": {
                "line": 5,
                "column": 8
            }
        }
    },
    "value": {
        "type": "FunctionExpression",
        "id": null,
        "generator": false,
        "expression": false,
        "async": false,
        "body": {
            "type": "BlockStatement",
            "body": [
                {
                    "type": "ReturnStatement",
                    "argument": {
                        "type": "CallExpression",
                        "callee": {
                            "type": "MemberExpression",
                            "object": {
                                "type": "Identifier",
                                "name": "Promise",
                                "range": [138, 145],
                                "loc": {
                                    "start": {
                                        "line": 6,
                                        "column": 11
                                    },
                                    "end": {
                                        "line": 6,
                                        "column": 18
                                    }
                                }
                            },
                            "property": {
                                "type": "Identifier",
                                "name": "resolve",
                                "range": [146, 153],
                                "loc": {
                                    "start": {
                                        "line": 6,
                                        "column": 19
                                    },
                                    "end": {
                                        "line": 6,
                                        "column": 26
                                    }
                                }
                            },
                            "computed": false,
                            "range": [138, 153],
                            "loc": {
                                "start": {
                                    "line": 6,
                                    "column": 11
                                },
                                "end": {
                                    "line": 6,
                                    "column": 26
                                }
                            }
                        },
                        "arguments": [
                            {
                                "type": "Literal",
                                "raw": "'foo'",
                                "value": "foo",
                                "range": [154, 159],
                                "loc": {
                                    "start": {
                                        "line": 6,
                                        "column": 27
                                    },
                                    "end": {
                                        "line": 6,
                                        "column": 32
                                    }
                                }
                            }
                        ],
                        "range": [138, 160],
                        "loc": {
                            "start": {
                                "line": 6,
                                "column": 11
                            },
                            "end": {
                                "line": 6,
                                "column": 33
                            }
                        }
                    },
                    "range": [131, 161],
                    "loc": {
                        "start": {
                            "line": 6,
                            "column": 4
                        },
                        "end": {
                            "line": 6,
                            "column": 34
                        }
                    }
                }
            ],
            "range": [125, 165],
            "loc": {
                "start": {
                    "line": 5,
                    "column": 11
                },
                "end": {
                    "line": 7,
                    "column": 3
                }
            }
        },
        "range": [122, 165],
        "params": [],
        "loc": {
            "start": {
                "line": 5,
                "column": 8
            },
            "end": {
                "line": 7,
                "column": 3
            }
        }
    },
    "computed": false,
    "static": false,
    "kind": "method",
    "range": [92, 165],
    "loc": {
        "start": {
            "line": 4,
            "column": 2
        },
        "end": {
            "line": 7,
            "column": 3
        }
    },
    "decorators": [
        {
            "type": "Decorator",
            "expression": {
                "type": "CallExpression",
                "callee": {
                    "type": "Identifier",
                    "name": "SomeFancyDecorator",
                    "range": [93, 111],
                    "loc": {
                        "start": {
                            "line": 4,
                            "column": 3
                        },
                        "end": {
                            "line": 4,
                            "column": 21
                        }
                    }
                },
                "arguments": [],
                "range": [93, 113],
                "loc": {
                    "start": {
                        "line": 4,
                        "column": 3
                    },
                    "end": {
                        "line": 4,
                        "column": 23
                    }
                }
            },
            "range": [92, 113],
            "loc": {
                "start": {
                    "line": 4,
                    "column": 2
                },
                "end": {
                    "line": 4,
                    "column": 23
                }
            }
        }
    ]
}
```

</details>

**Comment by SuperOleg39** [comment]
Hello!

Behaviour, when Node range include decorators range - default Typescript AST behaviour, is inconsistent with babel-parser behaviour, for example you can see this issue https://github.com/babel/babel/issues/6676

Maybe when converting we need to reset ranges?

**Comment by bradzacher** [comment]
please file a new issue instead of commenting on an old, closed issue.
https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md#commenting
